coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/loader/sql/SqlLoader.scala

```diff
@@ -221,21 +271,37 @@ final class SqlLoader[CT, CV](val connection: Connection,
     maybeFlush()
   }
 
-  def delete(jobId: Int, id: CV, version: Option[Option[RowVersion]]): Unit = {
+  def delete(jobId: Int, id: CV, version: Option[Option[RowVersion]], bySystemId: Boolean): Unit = {
     checkJob(jobId)
-    if(currentBatch.hasUpsertFor(id)) {
-      log.debug("Delete forced a flush; potential pipeline stall")
+
+    val deleteBySystemId = datasetContext.hasUserPrimaryKey && bySystemId
```

Comment:
Should this be `bySystemId || !datasetContext.hasUserPrimaryKey`?

Reply:
no, but I should change the variable name to `deleteBySystemIdForced` or something

Reply:
it is doing what i intended...

Reply:
:+1: